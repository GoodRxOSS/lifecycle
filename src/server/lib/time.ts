/**
 * Copyright 2025 GoodRx, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import moment, { Moment } from 'moment';

export function getTimestamp(
  time: Moment | string,
  format = 'YYYY-MM-DD HH:mm:ss'
) {
  const instance = moment.isMoment(time) ? time : moment(time);
  return instance.format(format);
}

export function getUtcTimestamp(time: string | Moment = moment()) {
  return getTimestamp(moment.utc(time));
}

export function isTimestampExpired(
  date: string | Moment,
  now: string | Moment = moment()
) {
  return moment.utc(now).isAfter(moment.utc(date));
}
