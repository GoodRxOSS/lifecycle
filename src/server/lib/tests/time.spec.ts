/**
 * Copyright 2025 GoodRx, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import moment from 'moment';
import * as timeUtils from '../time';

describe('Time utils', () => {
  describe('#getTimestamp', () => {
    const testDate = '2019-07-04';
    test('returns formatted date', () => {
      expect(timeUtils.getTimestamp(testDate)).toMatchSnapshot();
    });

    test('takes custom formatting string', () => {
      expect(timeUtils.getTimestamp(testDate, 'MM/DD/YYYY')).toMatchSnapshot();
    });
  });

  describe('#isTimestampExpired', () => {
    test('falsy if provided date is in the future', () => {
      const futureDate = moment().add(30, 'seconds');
      expect(timeUtils.isTimestampExpired(futureDate)).toBeFalsy();
    });

    test('truthy if provided date is in the past', () => {
      const futureDate = moment().subtract(30, 'seconds');
      expect(timeUtils.isTimestampExpired(futureDate)).toBeTruthy();
    });
  });
});
