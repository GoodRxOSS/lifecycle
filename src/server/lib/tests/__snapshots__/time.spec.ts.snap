// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time utils #getTimestamp returns formatted date 1`] = `"2019-07-04 00:00:00.0000"`;

exports[`Time utils #getTimestamp takes custom formatting string 1`] = `"07/04/2019"`;
